config_ext_test.go

```diff
@@ -0,0 +1,350 @@
+package config_test
+
+import (
+	"strings"
+	"testing"
+
+	. "go.uber.org/config"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIntegration(t *testing.T) {
+	first := strings.NewReader(`
+nothing: ~
+fun:
+  - maserati
+  - porsche
+practical:
+  toyota: camry
+  honda: accord
+occupants:
+  honda:
+    driver: jane
+    backseat: [nate]
+`)
+	second := strings.NewReader(`
```

Comment:
can we also add a test where a value in `first` is set to `~` in `second`, ensuring that overrides to `nil` work

Reply:
Done.